docs: blog post about graphback (#266)
aerogear/graphback: website/blog/2019-08-21-introduction.md

Answers for these questions led us to the creation of the new GraphQL based ecosystem called Graphback.

## Introducing Graphback

![Alt Text](https://thepracticaldev.s3.amazonaws.com/i/b58zov9fctzk44qzwrln.png)

The target of Graphback is to make sample applications working with your business logic by utilizing application templates and code generation. Developers can provide GraphQL schema as input and the command-line client will generate data access methods on top of the chosen stack (called template) and databases we support. 

Graphback generates Node.JS application with an entire codebase that is editable and can be later deployed to your own server or cloud provider.
By default, Graphback will follow the best patterns and include proven open-source libraries provided by the vibrant GraphQL community. When working with relational databases Graphback will allow users to manage database schema. The real power of Graphback is its code generation engine that can be extended for different needs. 

![diagram](https://thepracticaldev.s3.amazonaws.com/i/c9q6rvbb9v534073cq6j.png)

## Graphback to resolve major issues in GraphQL community

> Graphback helps developers to boost their getting started the experience with GraphQL. Graphback CLI offers different workflows supporting fetching data from the database and rest endpoints. 

Reviewer: REST not rest?

> Graphback helps developers to boost their getting started the experience with GraphQL. Graphback CLI offers different workflows supporting fetching data from the database and rest endpoints. 

Reviewer 2: ```suggestion
Graphback helps developers to boost their getting started experience with GraphQL. The Graphback CLI offers different workflows supporting data retrieval from the database and REST endpoints.
```

> Templates allow having different flavors of the GraphQL frameworks like Relay, Apollo, and Prisma for database access. Thanks to templates developers can avail of a fully functional set of the libraries that produce business value and work really well together. The community can provide their own templates to contribute to the success of the GraphQL entire ecosystem. 

Reviewer 2: ```suggestion
Templates allow having different flavors of the GraphQL frameworks like Relay, Apollo, and Prisma for database access. Thanks to templates developers can avail of a fully functional set of libraries that produce business value and work really well together. The community can provide their own templates to contribute to the success of the entire GraphQL ecosystem.
```

Reviewer 2: Should "flavors" be "flavours"? It depends on if we want US English or Irish/British.

Graphback generated application source code will be always in your full control - no hidden dependencies or any additional limitations included from Graphback runtime. Graphback will generate CRUD data access methods that can be extended into more customized data access functions at any time.

> Developers can introduce custom queries and mutations that will generate data access stubs. Thanks to first-class TypeScript support generated projects offer type safety.  

Reviewer 2: ```suggestion
Developers can introduce custom queries and mutations that will generate data access stubs. Thanks to first-class TypeScript support, generated projects offer type safety.
```

## Ecosystem of templates

> Graphback builds an ecosystem for developers that can build sample apps as templates that can be extended by additional features and business logic. Graphback templates follow canonical implementations and allow developers to continue development traditional way and tweak every aspect of the application. 

Reviewer 2: ```suggestion
Graphback builds an ecosystem for developers that can build sample apps as templates that can be extended by additional features and business logic. Graphback templates follow canonical implementations and allow developers to continue development in a traditional way and tweak every aspect of the application.
```

![Alt Text](https://thepracticaldev.s3.amazonaws.com/i/ss0zueqh885m8h5wra1y.gif)

> ## Not only GraphQL Server

Reviewer 2: ```suggestion
## Not just a GraphQL server
```

> Apart from fully functional backend server developers can choose to generate GraphQL queries for the client-side applications. Client-side code can utilize form generators like https://uniforms.tools giving developers end to end front-end and back-end solutions that will rely on their schema. 

Reviewer 2: ```suggestion
Apart from fully functional back-end server, developers can choose to generate GraphQL queries for client-side applications. Client-side code can utilize form generators like https://uniforms.tools, giving developers end to end front-end and back-end solutions that will rely on their schema. 
```

## Support 

> Our project is maintained by volunteers who are devoted to constant maintenance. 

Reviewer 2: ```suggestion
Our project is maintained by volunteers who are devoted to constant love and maintenance. 
```

> You can support us by staring project on GitHub or posting on social media

Reviewer 2: ```suggestion
You can support us by becoming a [Graphback stargazer](https://github.com/aerogear/graphback/stargazers) on GitHub, or by sharing it on social media.
```

> Our github: 

Reviewer 2: ```suggestion
Our GitHub: 
```
